hazelcast/test/src/sql_test.cpp

```diff
@@ -505,6 +866,392 @@ TEST_F(SqlTest, statement_with_params)
+         map_name)
+          .str()
+    };
+
+    statement.timeout(std::chrono::milliseconds{ 100 });
+
+    auto expected_row_count = count_if(
+      begin(expecteds), end(expecteds), [](const test::student& student) {
+          return student.height < 100.0f;
+      });
+
+    auto result = client.get_sql().execute(statement).get();
+
+    for_each_row(result,
+                 assert_row_count(expected_row_count),
+                 [&expecteds](const sql::sql_page::sql_row& row) {
+                     auto idx = row.get_object<int>(0);
+
+                     EXPECT_TRUE(idx.has_value());
+                     auto height = row.get_object<float>(1);
+                     EXPECT_TRUE(height.has_value());
+                     EXPECT_EQ(expecteds.at(*idx).height, *height);
+                 });
+}
+
+TEST_F(SqlTest, test_statement_with_wrong_timeout)
+{
+    sql::sql_statement statement{ client, "non_null_query" };
+
+    ASSERT_THROW(statement.timeout(std::chrono::milliseconds{ -500 }),
+                 exception::illegal_argument);
+}
+
+TEST_F(SqlTest, test_statement_with_empty_sql)
+{
+    auto fn = [this]() { sql::sql_statement statement{ client, "" }; };
+
+    ASSERT_THROW(fn(), exception::illegal_argument);
+}
+
+TEST_F(SqlTest, test_execute_with_cursor_buffer_size)
+{
+    static constexpr int CURSOR_BUFFER_SIZE = 3;
+
+    create_mapping_for_student();
+    (void)populate_map<test::student>(48);
+
+    sql::sql_statement statement{
+        client, (boost::format("SELECT * FROM %1%") % map_name).str()
+    };
+
+    statement.cursor_buffer_size(CURSOR_BUFFER_SIZE);
+
+    auto result = client.get_sql().execute(statement).get();
+
+    for (auto itr = result->iterator(); itr.has_next();)
+        EXPECT_EQ(itr.next().get()->row_count(), CURSOR_BUFFER_SIZE);
+}
+
+TEST_F(SqlTest, test_execute_with_schema)
+{
+    create_mapping_for_student();
+    (void)populate_map<test::student>(48);
+
+    sql::sql_statement statement{
+        client, (boost::format("SELECT * FROM %1%") % map_name).str()
+    };
+
+    statement.schema(std::string{ "nothing" });
+
+    client.get_sql().execute(statement).get();
+}
+
+TEST_F(SqlTest, test_execute_with_expected_result_type)
+{
+    create_mapping_for_student();
+    (void)populate_map<test::student>();
+
+    sql::sql_statement statement{
+        client, (boost::format("SELECT * FROM %1%") % map_name).str()
+    };
+
+    statement.expected_result_type(sql::sql_expected_result_type::rows);
+
+    client.get_sql().execute(statement).get();
+}
+
+TEST_F(SqlTest, test_is_row_set_when_row_is_set)
+{
+    create_mapping_for_student();
+    auto expecteds = populate_map<test::student>(100);
+
+    sql::sql_statement statement{
+        client, (boost::format("SELECT * FROM %1%") % map_name).str()
+    };
+
+    auto result = client.get_sql().execute(statement).get();
+
+    ASSERT_TRUE(result->row_set());
+}
+
+TEST_F(SqlTest, test_is_row_set_when_there_is_no_update)
```

Comment:
How is this test different from above `test_is_row_set_when_row_is_set ` ?

Reply:
They do the same execution but they assert different aspects of the `sql_result`

Reply:
Decision: Change the second one sql statement to use update which returns update count 0.

Reply:
Fixed.